libsolidity/ast/Types.cpp

```diff
@@ -147,6 +147,21 @@ bool fitsIntoBits(bigint const& _value, unsigned _bits, bool _signed)
 	));
 }
 
+TypePointers transformParametersToExternal(TypePointers const& _parameters, bool _inLibrary)
+{
+	TypePointers transformed;
+
+	for (auto type: _parameters)
+	{
+		if (auto ext = type->interfaceType(_inLibrary).get())
+			transformed.push_back(ext);
+		else
+			return {};
```

Comment:
Callers of this function cannot easily distinguish between an error case and an invocation on the empty vector. Why not use a `Result<TypePointers>` and also hand over the error message from the `interfaceType` call?